packages/modules/providers/payment-stripe/src/core/stripe-base.ts

```diff
@@ -272,36 +250,114 @@ abstract class StripeBase extends AbstractPaymentProvider<StripeOptions> {
   async updatePayment(
     input: UpdatePaymentProviderSession
   ): Promise<PaymentProviderError | PaymentProviderSessionResponse> {
-    const { context, data, currency_code, amount } = input
+    const { data, currency_code, amount } = input
 
     const amountNumeric = getSmallestUnit(amount, currency_code)
+    if (isPresent(amount) && data.amount === amountNumeric) {
+      return { data }
+    }
 
-    const stripeId = context.customer?.metadata?.stripe_id
+    try {
+      const id = data.id as string
+      const sessionData = (await this.stripe_.paymentIntents.update(id, {
+        amount: amountNumeric,
+      })) as unknown as PaymentProviderSessionResponse["data"]
 
-    if (stripeId !== data.customer) {
-      return await this.initiatePayment(input)
-    } else {
```

Comment:
question: Don't we need to deal with this scenario still?
1. I initialize my cart as a guest customer
2. Go through the checkout flow and create payment sessions
3. I sign in -> should update the customer on the payment intent

Reply:
Actually, I am not sure our workflows are wired up to deal with this right now.

Reply:
They are not, and this method is not even called.

I was gonna ask you why https://github.com/medusajs/medusa/pull/9348/files was closed actually, as I imagine that's where these flows would be implemented